src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SemanticFacts/CSharpSemanticFacts.cs

```diff
@@ -83,6 +83,9 @@ public bool CanReplaceWithRValue(SemanticModel semanticModel, [NotNullWhen(true)
                 // If we hit an executable statement syntax and didn't find anything yet, we can just stop now -- anything higher would be a member declaration which won't be defined by something inside a statement.
                 if (CSharpSyntaxFacts.Instance.IsExecutableStatement(ancestor))
                     return null;
+
+                if (CSharpSyntaxFacts.Instance.IsIdentifierContainerPreprocessorDirectiveTrivia(ancestor))
+                    return semanticModel.Compilation.CreatePreprocessingSymbol(token.ValueText);
```

Comment:
i don't undrestand this.

Reply:
This determines that the given token represents a preprocessing symbol identifier, since one of its ancestors is a preprocessor identifier container directive trivia through the helper method.

Reply:
this is GetDeclaredSymbol.  So this should only apply for an #if directive.  I'm also not sure why this semantic fact is even needed in this PR.  Where is it used?

Reply:
When getting the symbol that is to be sought in the reference finder, this is the path that is taken. The behavior about which directive trivia kinds should be supported and which ones should not be could change, according to a more mature design around supporting getting preprocessing symbol info.